matching-service/model/repository.js

```diff
@@ -0,0 +1,35 @@
+import 'dotenv/config';
+import MatchModel from './match-model.js';
+
+//Set up mongoose connection
+import mongoose from 'mongoose';
+
+let mongoDB = process.env.ENV == 'PROD' ? process.env.DB_CLOUD_URI : process.env.DB_LOCAL_URI;
+
+mongoose.connect(mongoDB, { useNewUrlParser: true, useUnifiedTopology: true });
+
+let db = mongoose.connection;
+db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+
+export async function createMatch(params) {
+    return new MatchModel(params);
+}
+
+export async function findMatch(params) {
+    const result = MatchModel.findOne({
+        $or: [
+            { username1: params.username, isActive: true },
+            { username2: params.username, isActive: true }
+        ]
+    });
+
+    return result;
+}
+
+export async function endMatch(params) {
+    const result = MatchModel.findOneAndUpdate({ username1: params.username, isActive: true }, { isActive: false });
+    if (result) {
+        return result;
+    }
+    return MatchModel.findOneAndUpdate({ username2: params.username, isActive: true }, { isActive: false });
```

Comment:
can the `$or` operator also be used for `findOneAndUpdate` ? also i think it might help to handle the case where there does not exist an entry with the given parameters with a try-catch here, same for `findMatch`.

Reply:
yes I think it should work. I will update this. Regarding the second point, if there does not exist such entry, it will simply return null. I don't think there is a need to classify this as an error. other parts handles the case when the result is null.